### PR TITLE
Add {set,get}_bucket_versioning() APIs

### LINKED ISSUE #974: Implement GetBucketVersioning API
We need to implement GetBucketVersioning API in minio-py like it is done in https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/BucketVersioningConfiguration.html

### PR DESCRIPTION
**Pull request by balamurugana**
Fixes #974

### CODE REVIEW

**Review by ebozduman — COMMENTED** [review]
Lots of changes. So, this is a major refactoring job.
Probably it would be a good idea to change the title;
`Add {set,get}_bucket_versioning() APIs`

**Comment by balamurugana** [comment]
> Lots of changes. So, this is a major refactoring job addressing 2 issues; #974 and ? (refactoring task of putting http execution at one place).
> Probably it would be a good idea to change the title;
> `Add {set,get}_bucket_versioning() APIs`

As mentioned in the description (first comment), this PR Depends on #968

File: minio/api.py [hunk 1]
```diff
@@ -169,6 +156,323 @@ def __init__(self, endpoint, access_key=None,
...
+        on failure.
+        """
+        code, message = {
+            301: ("PermanentRedirect", "Moved Permanently"),
+            307: ("Redirect", "Temporary redirect"),
+            400: ("BadRequest", "Bad request"),
+        }.get(response.status, (None, None))
+        region = response.getheader("x-amz-bucket-region")
+        if message and region:
+            message += "; use region " + region
```
[the remark this answers is not in the document]
Comment (balamurugana) [reply minio/api.py:173]: No. S3 service returns an error with new region, hence the message is correct.

**Review by ebozduman — APPROVED** [review]
LGTM

**Review by BigUstad — APPROVED** [review]
LGTM

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**